### DISCUSSION TITLE
HX711 issue

### DISCUSSION THREAD

**Discussion by rokbass** [micropython/micropython#9836] [Core Development]
Hello,  New to Github and micropython.  I have done a lot of reading and research on this site, but I can't seem to find the exact cause of my dilemma.  I am using a raspberry pi pico.  programming with Thonny ide.  I have successfully wired and programmed an oled display using the tom's hardware example in thonny.  Now, I want to add a load cell using an HX711.  I should add that I have successfully done this using the Arduino IDE.  I'm not a pro by any means, but I really want to learn micropython better.
I have kept the simply code for the oled in, as the whole point of this simply program is to display load cell info.  I am using the HX711 for micropython library.  
I have entered this code line by line, and only get the error in line 14.  My thinking is that there is something wrong with the library, but this is up to date and is really the only one I can find specific to micropython and RP2040.  Any help would be appreciated.
![thonny issue](https://user-images.githubusercontent.com/117464950/200018474-e7efdeb5-dbce-454a-85b4-dd862e50910e.png)

**Comment by robert-hh** [comment]
The error tells, that the HX711 constructor expects integers, but you supply pin objects. Which HX711 library do you use? Please tell us the url of that library.

**Reply by rokbass** [reply]
https://github.com/SergeyPiskunov/micropython-hx711

I thought the same thing.  
So then I changed to 
hx711 = HX711(4, 5).  
That worked, but then no further commands were being recognized referencing the hx711.  Should I be using a different library?
Thanks so much.

**Reply by robert-hh** [reply]
The lib looks ok so far. If you do not get anything returned, check the wiring. One pitfall of the RP2 pico is the pin number scheme. The code refers to GPIO numbers, not board pin numbers. GPIO4 is board pin number 6, and GPIO5 is board pin number 7.

**Reply by scruss** [reply]
Yes, the Pico's number-only system has its limitations.

This (new) website is quite helpful: [Raspberry Pi Pico GPIO Pinout](https://pico.pinout.xyz/)

**Reply by rokbass** [reply]
Thank you very much for the quick replies to this.  It wasn't an issue with GPIO numbering, it turns out that I needed to use the hx711_gpio library instead of the one I had listed above.  Once I changed the library, the code worked.  However, that only brings up another question.  Did the new library have a string to int conversion?  If so, I couldn't see it.  Long story short, the error disappeared after loading the new library, and everything works great now!  Thanks again for your help.

**Reply by robert-hh** [reply]
There is the standard conversion method int(obj). So int("123") returns 123.

**Comment by v-srirama** [comment]
Hi @rokbass , do you mind sharing the "hx711_gpio"'s GitHub page? I cannot find it

**Reply by robert-hh** [reply]
I have a hx711 driver version of that name here: https://github.com/robert-hh/hx711

**Reply by v-srirama** [reply]
Thank you!
